solutions/usaco-787.mdx

```diff
@@ -87,4 +86,81 @@ int main() {
 ```
 
 </CPPSection>
+<JavaSection>
+```java
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.StringTokenizer;
+
+public class rental {
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new FileReader("rental.in"));
+        PrintWriter out = new PrintWriter("rental.out");
+        StringTokenizer s = new StringTokenizer(br.readLine());
+        int n = Integer.parseInt(s.nextToken());
+        int m = Integer.parseInt(s.nextToken());
+        int r = Integer.parseInt(s.nextToken());
+        Integer [] production = new Integer [n];
+        for (int i = 0; i < n; i++) {
+            production[i] = Integer.parseInt(br.readLine());
+        }
+        Integer [][] shops = new Integer [m][2];
+        for (int i = 0; i < m; i++) {
+            s = new StringTokenizer(br.readLine());
+            shops[i][1] = Integer.parseInt(s.nextToken());
+            shops[i][0] = Integer.parseInt(s.nextToken());
+        }
+        Integer [] rent = new Integer [r];
+        for (int i = 0; i < r; i++) {
+            rent[i] = Integer.parseInt(br.readLine());
+        }
+        Arrays.sort(rent, Collections.reverseOrder());
+        Arrays.sort(shops, new Comparator<Integer[]>() {
+            @Override
+            public int compare(Integer[] a, Integer[] b) {
+                return b[0] - a[0];
+            }
+        });
+        Arrays.sort(production, Collections.reverseOrder());
+        int iShop = 0;
+        int iRent = 0;
+        int i = 0;
+        long sol = 0;
+        while (i < n) {
+            int litres = production[i];
+            int moneyT = 0, curI = iShop;
```

Comment:
It's not too clear what `moneyT` and `curI` mean, maybe rename them? Again, up to you.

Reply:
or you can write a comment next to the variable to indicate what it is supposed to mean if you want to keep the variable name.

Reply:
Uh I just copied the variable names from the existing c++ solution in order for the java and c++ solutions to be consistent.  I'll just add comments.

Reply:
Sounds good, you can disregard my comments then